src/mmtk.rs

```diff
@@ -597,3 +597,13 @@ impl<VM: VMBinding> MMTK<VM> {
             .initialize_object_metadata(object, false)
     }
 }
+
+#[no_mangle]
+pub fn mmtk_debug_print_object_info(object: crate::util::ObjectReference) {
```

Comment:
I think this function is intended to be called from C or GDB.  In this case, the user may just supply an arbitrary number and want it to be used as an `ObjectReference`.  But `ObjectReference` is not `repr(C)`, and it has validity requirements (non-zero and word-aligned).  I think we can change this to `Address` and validate it inside this function.

```suggestion
pub fn mmtk_debug_print_object_info(object_addr: crate::util::Address) {
    let Some(object) = ObjectReference::from_raw_address(object_addr) else {
        println!("{object_addr} is not a valid object reference");
        return;
    };
```

Reply:
By the way, if the `vo_bit` feature is enabled, we should use the VO bit to convert the object to `ObjectReference`.  But if that fails, we may still "force" converting `object_addr` to an `ObjectReference` to provide some information with "best-effort".